server/bin/pbench-verify-backup-tarballs.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+import sys
+import glob
+import hashlib
+import shutil
+import tempfile
+from argparse import ArgumentParser
+from s3backup import S3Config, Entry
+from pbench import init_report_template, report_status, _rename_tb_link, \
+    PbenchConfig, BadConfig, get_es, get_pbench_logger, md5sum
+
+_NAME_    = "pbench-verify-backup-tarballs"
+
+# Global logger for the module, setup in main()
+_logger = None
+
+def checkmd5(target_dir, reference_dir, indicator):
+    # Function to check integrity of results in a local (archive or local backup) directory
+    if not os.path.isdir(target_dir):
+        _logger.error('Bad {}: {}'.format(indicator, target_dir))
+        os._exit(1)
+    tarlist = glob.iglob(os.path.join(target_dir, "*", "*.tar.xz"))
+    with open(os.path.join(reference_dir, "list.{}".format(indicator)), 'w') as f_list:
+        for tar in tarlist:
+            result_name = os.path.basename(tar)
+            controller = os.path.basename(os.path.dirname(tar))
+            md5 = ("{}.md5".format(tar))
+            f_list.write("{}\n".format(os.path.join(controller, result_name)))
+            with open(md5) as f:
+                md5_value = f.readline().split(" ")[0]
+            md5_returned = md5sum(tar)
+            if md5_value == md5_returned:
+                with open(os.path.join(reference_dir, "list.{}.ok".format(indicator)), 'w') as f_ok:
+                    f_ok.write(
+                        "{}: {}\n".format(os.path.join(controller, result_name), "OK"))
+            else:
+                with open(os.path.join(reference_dir, "list.{}.fail".format(indicator)), 'w') as f_fail:
+                    f_fail.write(
+                        "{}: {}\n".format(os.path.join(controller, result_name), "FAILED"))
+
+
+def report_failed_md5(reference_dir, archive, backup, report):
```

Comment:
`archive` and `backup` are only used to write a report message that they are not mounted which you infer indirectly. It seems that that should be checked very early in the program (where we should bail if either is not mounted), not in this function. In factf, we *do* check: I think you should get rid of the two messages and the two arguments.